extensions/apicurio-registry-avro/runtime/src/main/java/io/quarkus/apicurio/registry/binding/ServiceRegistryBindingConverter.java

```diff
@@ -0,0 +1,107 @@
+
+    @Override
+    public Optional<ServiceBindingConfigSource> convert(List<ServiceBinding> serviceBindings) {
+        var matchingByType = ServiceBinding.singleMatchingByType("serviceregistry", serviceBindings);
+        Config config = ConfigProvider.getConfig();
+        if (matchingByType.isEmpty()) {
+            return Optional.empty();
+        }
+
+        var binding = matchingByType.get();
+
+        List<String> channels = extractChannels(config);
+
+        Map<String, String> properties = new HashMap<>();
+
+        String registryUrl = binding.getProperties().get("registryUrl");
+        if (registryUrl == null) {
+            registryUrl = binding.getProperties().get("registryurl");
+        }
+        if (registryUrl != null) {
+            properties.put("mp.messaging.connector.smallrye-kafka.apicurio.registry.url", registryUrl);
+        }
+        
+
+        for (String channel : channels) {
+
+            String prefix = channel;
+
+            String oAuthHost = binding.getProperties().get("oauthServerUrl");
+            if (oAuthHost == null) {
+                oAuthHost = binding.getProperties().get("oauthserverurl");
+            }
+            if (oAuthHost != null) {
+                properties.put(prefix + "apicurio.auth.service.url", oAuthHost);
+            }
+
+            String clientId = binding.getProperties().get("clientId");
+            if (clientId == null) {
+                clientId = binding.getProperties().get("clientid");
+            }
+            if (clientId != null) {
+                properties.put(prefix + "apicurio.auth.client.id", clientId);
+            }
+
+            String clientSecret = binding.getProperties().get("clientSecret");
+            if (clientSecret == null) {
+                clientSecret = binding.getProperties().get("clientsecret");
+            }
+            if (clientSecret != null) {
+                properties.put(prefix + "apicurio.auth.client.secret", clientSecret);
+            }
+
+            String realm = binding.getProperties().get("oauthRealm");
+            if (realm == null) {
+                realm = binding.getProperties().get("oauthRealm");
+            }
+            if (clientSecret != null) {
+                properties.put(prefix + "apicurio.auth.realm", realm);
+            }
+
+            if (registryUrl != null) {
+                properties.put(prefix + "apicurio.registry.url", registryUrl);
+            }
+        }
+        
+        return Optional.of(new ServiceBindingConfigSource("serviceregistry-k8s-service-binding-source", properties));
+    }
+
+    private List<String> extractChannels(Config configIn) {
+
+        var list = new ArrayList<String>();
+
+        for (String propertyName : configIn.getPropertyNames()) {
+            if (propertyName.startsWith(INCOMING_PREFIX)) {
+                var channelName = propertyName.replace(INCOMING_PREFIX, "").split("\\.")[0];
```

Comment:
This part looks risky to do with split potentially returning empty array in some circumastances

Reply:
Yeah, the channel name detection is a bit of a hack; maybe the service registry team could have a though on how to do this more elegantly?

Reply:
I guess we can check size of array before doing [0]
```suggestion
                var channelArray = propertyName.replace(INCOMING_PREFIX, "").split("\\.")
                if( channelArray. length()>0){
                     var channelName = channelArray[0];
                }
```

Reply:
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#substringAfter-java.lang.String-int- is probably safer